src/deepsparse/transformers/haystack/nodes.py

```diff
@@ -298,7 +295,15 @@ def embed_documents(self, docs: List[Document]) -> List[numpy.ndarray]:
         :param docs: list of document strings to embed
         :return: list of embeddings for each document
         """
-        passage_inputs = [doc.content for doc in docs]
+        passage_inputs = [
+            f"{doc.meta['title']}{self.title_separator}{doc.content}"
```

Comment:
i'd recommend moving this logic to a method, maybe like

```python
def doc_to_passage(self, doc): # you can probably think of a better name for this function haha
    ...
```

As this allows you to unit test the functionality and probably will be formatted slightly better

Reply:
Makes sense, I'll make that change